src/cockpit/peer.py

```diff
@@ -88,6 +92,13 @@ def _connect_task_done(task: asyncio.Task) -> None:
         connect_task = asyncio.create_task(self.do_connect_transport())
         connect_task.add_done_callback(_connect_task_done)
 
+        async def remove_auth_responses_after_two_minutes():
+            await asyncio.sleep(120)
+            self.auth_challenge_responses = None
+
+        if self.auth_challenge_responses:
+            self.auth_challenge_responses_remove_task = asyncio.create_task(remove_auth_responses_after_two_minutes())
```

Comment:
This is overkill.  I think you want https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.loop.call_later instead.  It returns a `TimerHandle` which you can use for cancellation.